### ISSUE TITLE
Module not context aware

### ISSUE THREAD

**Issue by duro** [geekyme/react-doc-meta#2]
I am running into an issue with meta tags duplicating when rendered on the server. Since the module exports as a singleton, the `_serverMeta` array keeps appending the same tags to it with every page load.

Am I missing something or is there some way to make this component context aware.

**Comment by geekyme** [comment]
I can't reproduce your issue. 

Have you called ```DocMeta.rewind()``` on the server side?

for instance on the server side HtmlComponent I am doing something like this

```
...
    let metaTags = DocMeta.rewind();

    return (
        <head>
          <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
          {
            metaTags.map((tag, index) =>
              <meta data-doc-meta="true" key={index} {...tag} />)
          }

        </head>

        <body>
          <div id="root" dangerouslySetInnerHTML={{__html: markup}} />
        </body>
      </html>
    );
...
```

**Comment by duro** [comment]
That did not solve it.

I have tried using a server side template (via Hapi) to render my HTML, as well as used `React.renderToStaticMarkup`.

With both cases, the meta data is duplicated with each page request.

I have inspected the react-doc-meta module in my node-inspector and set a breakpoint on L47. If I inspect the `_serverMeta` array, I can see the data from the previous request still in there.

I believe this is becuase the `_serverMeta` array is set to module level scope. Since modules are singletons, they are re-used for as long as the application is running. This means that array is never reset.

**Comment by geekyme** [comment]
Does https://github.com/gaearon/react-document-title work for you? 

I'm using the same approach as that module to set side effects.

**Comment by geekyme** [comment]
is there some way i can reproduce this though? I cannot see this happening even when I'm firing 200 rps to my server

**Comment by geekyme** [comment]
hey @duro managed to reproduced the issue already. The ```_serverMeta``` keeps growing and causes a memory leak. I'll accept any PR if u have a fix.

**Comment by xpepermint** [comment]
Also seeing this. +1

**Comment by duro** [comment]
I was able to hack a solution into place for my project, but I really don't think it's the "right" way.

Here is my modified component:

https://gist.github.com/duro/5c7d5b7d7d4f6bba2fee

Note that for this to work, I had to add an "ID" to the component implementation:

```html
<DocMeta tags={meta} id="home" />
```

The reason for this is that in some cases my components go through more than one render on the server. This ID acts as a unique key for the page, so that when it get's added to _serverMeta it replaces the same key if a double render occurs.

Also, here is the snippet I am using to parse through the returned _serverMeta that comes back from DocTitle.rewind() to get these tags into the HTML template (I am not using a React component to render my HTML template)

```javascript
locals.metas = [];
_.forIn(metaData, (meta, pageKey) => {
  meta.forEach((m) => {
    var metaStrings = []
    _.forIn(m, (value, key) => {
      metaStrings.push(key + '="' + value + '"');
    });
    metaStrings.push('data-doc-meta="true"');
    locals.metas.push("<meta " + metaStrings.join(" ") + " />");
  });
});
```

I'm sure I could clean this up, but I wanted this done so I could move on.
